From 9ef6be5ae6245ad7b0b1b567a5e2d44e1e661776 Mon Sep 17 00:00:00 2001
From: Howard Hsu <howard-yh.hsu@mediatek.com>
Date: Tue, 19 Nov 2024 20:35:02 +0800
Subject: [PATCH 02/95] mtk: mt76: mt7996: fix thermal notify event

Fix thermal notify event tag and event id because of firmware change.
Additionally, remove one unused entry point receiving thermal notify
event.

Signed-off-by: Howard Hsu <howard-yh.hsu@mediatek.com>
---
 mt7996/mcu.c | 8 ++------
 1 file changed, 2 insertions(+), 6 deletions(-)

diff --git a/mt7996/mcu.c b/mt7996/mcu.c
index f0adc0b4..53fdf86f 100644
--- a/mt7996/mcu.c
+++ b/mt7996/mcu.c
@@ -598,8 +598,7 @@ mt7996_mcu_rx_all_sta_info_event(struct mt7996_dev *dev, struct sk_buff *skb)
 static void
 mt7996_mcu_rx_thermal_notify(struct mt7996_dev *dev, struct sk_buff *skb)
 {
-#define THERMAL_NOTIFY_TAG 0x4
-#define THERMAL_NOTIFY 0x2
+#define THERMAL_NOTIFY_TAG 0x6
 	struct mt76_phy *mphy = &dev->mt76.phy;
 	struct mt7996_mcu_thermal_notify *n;
 	struct mt7996_phy *phy;
@@ -609,7 +608,7 @@ mt7996_mcu_rx_thermal_notify(struct mt7996_dev *dev, struct sk_buff *skb)
 	if (le16_to_cpu(n->tag) != THERMAL_NOTIFY_TAG)
 		return;
 
-	if (n->event_id != THERMAL_NOTIFY)
+	if (n->event_id != THERMAL_NOTIFY_TAG)
 		return;
 
 	if (n->band_idx > MT_BAND2)
@@ -646,9 +645,6 @@ mt7996_mcu_rx_unsolicited_event(struct mt7996_dev *dev, struct sk_buff *skb)
 	case MCU_EVENT_EXT:
 		mt7996_mcu_rx_ext_event(dev, skb);
 		break;
-	case MCU_UNI_EVENT_THERMAL:
-		mt7996_mcu_rx_thermal_notify(dev, skb);
-		break;
 	default:
 		break;
 	}
-- 
2.45.2

